grpc-api/src/main/java/io/deephaven/grpc_api/barrage/BarrageStreamGenerator.java

```diff
@@ -315,25 +310,22 @@ private InputStream getInputStream(final SubView view) throws IOException {
             final ChunkInputStreamGenerator.BufferListener bufferListener =
                     (offset, length) -> bufferInfos.add(new ChunkInputStreamGenerator.BufferInfo(offset, length));
 
-            for (int i = addColumns.original.nextSetBit(0), j = 0; i != -1; i = addColumns.original.nextSetBit(i + 1), j++) {
-                if (!myAddColumns.get(i)) {
-                    continue;
+            // add the add-column streams
+            for (int i = 0; i < addColumnData.length; ++i) {
```

Comment:
Write up an issue to interact with the arrow team and suggest alternatives to sending column data for empty columns. This would be _*expensive*_ in some real-world use cases.